src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -265,38 +266,44 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
     * @param listCon the list of Contacts to update
     * @return void  
     ********************************************************************************************************/
-    private void setConAddrFromLookup(list<Contact> listCon, DmlWrapper dmlWrapper, TDTM_Runnable.Action triggerAction) {
-    	
-    	list<Contact> listConCreateAddr = new list<Contact>();
-    	set<Id> setAddrId = new set<Id>();
-    	for (Contact con : listCon) {
-    		if (con.Current_Address__c != null)
-    		  setAddrId.add(con.Current_Address__c);
-    	}
-    
-        map<Id, Address__c> mapAddrIdAddr = new map<Id, Address__c>([select Id, Address_Type__c, Household_Account__c, 
-            MailingStreet__c, MailingStreet2__c, MailingCity__c, MailingState__c, Geolocation__Latitude__s, Geolocation__Longitude__s, 
-            MailingPostalCode__c, MailingCountry__c from Address__c where Id in :setAddrId]);
+    private void setConAddrFromLookup(List<Contact> listCon, DmlWrapper dmlWrapper, TDTM_Runnable.Action triggerAction) {
+
+        List<Contact> listConCreateAddr = new List<Contact>();
+        Set<Id> setAddrId = new Set<Id>();
+        for (Contact con : listCon) {
+            if (con.Current_Address__c != null) {
+                setAddrId.add(con.Current_Address__c);
+            }
+        }
+
+        Map<Id, Address__c> mapAddrIdAddr = new Map<Id, Address__c>([SELECT Id, Address_Type__c,
+                Household_Account__c, MailingStreet__c, MailingStreet2__c, MailingCity__c,
+                MailingState__c, Geolocation__Latitude__s, Geolocation__Longitude__s,
+                MailingPostalCode__c, MailingCountry__c
+            FROM Address__c
+            WHERE Id IN :setAddrId]
+        );
             
         //UTIL_Debug.debug('****DJH: setConAddrFromLookup mapAddrIdAddr: ' + mapAddrIdAddr);
             
         for (Contact con : listCon) {
             Address__c addr = mapAddrIdAddr.get(con.Current_Address__c);
-            if (addr != null) {
+            if (ADDR_Addresses_TDTM.doesAddressNeedUpdate(con, addr)) {
                 ADDR_Addresses_TDTM.copyAddressAddrSObj(addr, con, 'Mailing', 'npe01__Primary_Address_Type__c');
 
-            	// deal with any address coming from a different account,
-            	// which we need to treat as a creation instead.
-            	if (addr.Household_Account__c != con.AccountId) {
-            	   //UTIL_Debug.debug('****DJH address not on same account, so add to create list: con' + con);
-            	   listConCreateAddr.add(con);
-            	}
+                // deal with any address coming from a different account,
+                // which we need to treat as a creation instead.
+                if (addr.Household_Account__c != con.AccountId) {
+                    //UTIL_Debug.debug('****DJH address not on same account, so add to create list: con' + con);
```

Comment:
Fair point. I had left these behind just because they were there previously as commented out, but I can't really ever remember having to uncomment them out for any reason.